client/state/stats/chart-tabs/reducer.js

```diff
@@ -19,30 +19,43 @@ import { QUERY_FIELDS } from './constants';
  * @param  {Object} action Action payload
  * @return {Object}        Updated state
  */
-export function counts( state = {}, action ) {
-	switch ( action.type ) {
-		case STATS_CHART_COUNTS_RECEIVE: {
-			const ID = 'period';
-			const records = get( state, `${ action.siteId }.${ action.period }`, [] ).slice();
-			const recordIds = records.map( count => count[ ID ] );
+export const counts = keyedReducer(
+	'siteId',
+	keyedReducer( 'period', ( state = [], action ) => {
+		switch ( action.type ) {
+			case STATS_CHART_COUNTS_RECEIVE: {
+				const ID = 'period';
```

Comment:
It seems that there is no reason to have this `ID` variable. We can just find the index with simple:
```js
findIndex( nextState, { period: recordFromApi.period } )
```

Reply:
Fair enough, I'll write it explicitly. I'll keep using `Array.findIndex` instead of lodash's `findIndex`, though; no reason to pull that in for such a simple thing.